api/app/signals/apps/api/v1/private/views.py

```diff
@@ -195,3 +197,23 @@ def create(self, request, *args, **kwargs):
         self.perform_create(serializer)
 
         return self.retrieve(request, *args, **kwargs)
+
+
+class UserFilterViewSet(mixins.RetrieveModelMixin,
+                        mixins.ListModelMixin,
+                        mixins.CreateModelMixin,
+                        mixins.UpdateModelMixin,
+                        mixins.DestroyModelMixin,
+                        viewsets.GenericViewSet):
+    authentication_classes = (JWTAuthBackend,)
+    permission_classes = (SIAPermissions,)
+
+    pagination_class = HALPagination
+
+    serializer_class = UserFilterSerializer
+
+    def get_queryset(self):
+        from signals.apps.signals.models import Filter
+        return Filter.objects.filter(
+            Q(created_by=self.request.user.username) | Q(created_by__isnull=True)
```

Comment:
We talked about systemwide filters. So thought this could be some sort of first approach to it. Was just an idea that maybe we can build upon. Will remove it for now.

Reply:
agree